content/articles/creating-chatbot-using-natural-language-processing-in-python/index.md

```diff
@@ -0,0 +1,245 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /screating-chatbot-using-natural-language-processing-in-python/
+title: Creating ChatBot Using Natural Language Processing in Python
+description: In this article, will illustrate how to create a ChatBot using Natural Language Processing in Python. 
+author: vincent-kimanzi
+date: 2022-03-24T00:00:00-08:50
+topics: [Languages, API]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/screating-chatbot-using-natural-language-processing-in-python/hero.jpg
+    alt: Creating ChatBot Using Natural Language Processing in Python Hero Image
+---
+A bot is a computer program that performs predetermined tasks automatically. Its goal is to perform human duties the same way that humans do. In a nutshell, they mimic human behavior.
+<!--more-->
+In this tutorial, we will design a conversational interface for our chatbot using natural language processing.
+
+### Prerequisites
+- Have a basic knowledge of python programming langugage.
+- install the latest version of [python](https://www.python.org/) onto your computer.
+
+### Table of contents
+- [Fields of NLP](#fields-of-nlp)
+- [Types of bots](#types-of-bots)
+- [Building an NLP chatbot](#building-an-nlp-chatbot)
+- [Benefits of bots](#benefits-of-bots)  
+- [Conclusion](#conclusion)
+- [Reference](#reference)
+
+### Fields of NLP
+Below are the core fields of NLP:
+
+1. **Natural language generation(NLG)**: In artificial intelligence, NLG is a specialty. It is a software technology that automatically transforms data into simple English.
+2. **Natural language understanding(NLU)**: It is a branch of natural language processing (NLP) that helps computers understand and interpret human language by breaking speech into its constituent parts. However, NLU goes further than speech recognition to understand what the user is trying to communicate with their words.
+3. **Natural language interaction(NLI)**: To engage with any connected device or service in a humanlike manner, NLI brings together a varied collection of natural language principles.
+
+### Types of bots
+- **Scraper Bots:** They are designed to read and save data from the Internet.
+- **Spam Bots:** Designed to gather email addresses from spam mailing lists.
+- **Spider Bots:** Searches the web and downloads pages before indexing the material on each page it comes across.
+- **Transnational Bots:** Designed to be used in transactions.
+- **Social Media Bot:** Automatically responds to questions on social media platforms.
+
+### Building an NLP chatbot
+Follow the steps below to successfully build a conversational interface for our chatbot.
+
+#### Step 1: Importing modules
+Imports are critical for successfully organising your Python code. Importing code correctly will increase your productivity by allowing you to reuse code while also maintaining the maintainability of your projects.
+

```

Comment:
I wanted you to list down the modules you are importing.

Reply:
Okay I'll list them right now